### PR TITLE
Updating to latest schematics sdk and updating sch_agent resource to enable force deletion by default

### PR DESCRIPTION
**Pull request by VaishnaviGopal**
PS:
https://github.ibm.com/blueprint/product/issues/10753
https://github.ibm.com/blueprint/product/issues/10811
https://github.com/IBM-Cloud/terraform-provider-ibm/issues/5475

### CODE REVIEW

**Comment by VaishnaviGopal** [comment]
Screenshot 2024-09-20 at 2 25 36 AM

**Comment by hkantare** [comment]
fix the conflcits
Add testcase execution results

File: ibm/service/schematics/resource_ibm_schematics_agent_test.go [hunk 1]
```diff
@@ -195,9 +195,7 @@ func testAccCheckIbmSchematicsAgentDestroy(s *terraform.State) error {
 		// Try to find the key
 		_, response, err := schematicsClient.GetAgentData(getAgentDataOptions)
 		if err == nil {
-			return nil
-			// TODO: uncomment the following lines of code once the agent delete actually deletes agentdoc
-			// return fmt.Errorf("schematics_agent still exists: %s", rs.Primary.ID)
+			return fmt.Errorf("schematics_agent still exists: %s", rs.Primary.ID)
```
[the remark this answers is not in the document]
Comment (VaishnaviGopal) [reply ibm/service/schematics/resource_ibm_schematics_agent_test.go:198]: yes. This is exists scenario after deletion. Ideally it should be 404 after deletion. if it is not throwing any error that means deletion didnt happen and if it is throwing someother error other than 404, we are handling it in a different error string.

**Comment by VaishnaviGopal** [comment]
image
image
Screenshot 2024-09-20 at 2 25 36 AM

**Review by hkantare — APPROVED** [review]

**Pull request merged**